app/webpacker/components/CompetitionTabs/index.jsx

```diff
@@ -0,0 +1,95 @@
+import React, { useMemo } from 'react';
+import { Tab, TabPane } from 'semantic-ui-react';
+import GeneralInfoTab from './GeneralInfoTab';
+import CompetitionTab from './CompetitionTab';
+import EventsTable from './EventsTable';
+import Schedule from './Schedule';
+import WCAQueryClientProvider from '../../lib/providers/WCAQueryClientProvider';
+import './style.css';
+import TimeLimitCutoffInfo from './TimeLimitCutoffInfo';
+
+const updatePath = (tabSlug) => {
+  window.history.replaceState({}, '', `${window.location.pathname}#${tabSlug}`);
+};
```

Comment:
Do we have shared `lib` functions for this? I feel like one or two other parts of our frontend code use similar features already

Reply:
There is a function in filterUtils.js but it uses searchParams and not Hash so it works completely different